torch_spyre/csrc/spyre_mem.cpp

```diff
@@ -44,6 +48,411 @@
+                            layout, sendnn::TensorLocation::HOST());
+  sendnn::TensorInfo dev_ti(sendnnSpyreT, dev_tensor_shape, layout,
+                            sendnn::TensorLocation::DEVICE());
+  sendnn::TensorInfo dci_ti(sendnnSpyreT, dev_tensor_shape, layout,
+                            sendnn::TensorLocation::HOST());
+  //  STAGE 1: execution graph
+  sendnn::SubGraph fdc_graph;
+  int64_t xfer_size = dev_tensor_shape.Volume() * cpu_tensor->element_size();
+  {  // subgraph (execution graph)
+    flex::FlexGraphBuilder gb;
+    flex::FlexGraphBuilder* gb_sn = &gb;
+    DMAParameters tp{xfer_size, 0, 0};  // (num_bytes, offset_src, offset_dst)
+    if (host2device) {
+      auto pi = gb_sn->PrimaryInput("SN PI", dci_ti);
+      auto h2d_dt = gb_sn->SenDataTransfer(
+          "H2D DT",
+          dev_ti,  // output (holding shape, type, and location DEVICE)
+          pi,      // input (node created using PrimaryInput and on HOST)
+          dev_ti.DataSize(), tp.src_offset, tp.dst_offset);
+      auto po = gb_sn->PrimaryOutput("SN PO", h2d_dt);
+    } else {
+      auto pi = gb_sn->PrimaryInput("SN PI", dev_ti);
+      auto d2h_dt = gb_sn->SenDataTransfer(
+          "D2H DT",
+          dci_ti,  // output (holding shape, type and location HOST)
+          pi,  // input (node created as a result of SenDataTransfer node above,
+               // i.e. on DEVICE)
+          dev_ti.DataSize(),
+          tp.src_offset,   // device side
+          tp.dst_offset);  // host side
+      auto po = gb_sn->PrimaryOutput("SN PO", d2h_dt);
+    }
+
+    SEN_THROW_NOK(gb_sn->Finalize(&fdc_graph));
+  }
+  sendnn::SubGraph exec_graph;
+  {  // add above subgraph as part of SenFusedDeviceCompute node
+    flex::FlexGraphBuilder fgb;
+    if (host2device) {
+      auto inp_node = fgb.PrimaryInput("SN PI", cpu_ti);
+      auto h2d_dci = generate_dci(cpu_tensor, host2device);
+      auto h2d_dci_node = fgb.SenHostCompute(
+          "Host2Sen-HostPrep", {dci_ti}, {inp_node}, "SenDataConvert", h2d_dci);
+
+      auto fdc =
+          fgb.SenFusedDeviceCompute("FDC", {dci_ti}, {h2d_dci_node}, fdc_graph);
+      fgb.PrimaryOutput("SN PO", fdc->OutputPort(0));
+    } else {
+      sendnn::NodePtr inp_node = fgb.PrimaryInput(
+          "SN PI", dci_ti);  // not important in terms of shape (as it will be
+                             // empty when using)
+      // sendnn::NodePtr inp_node = fgb.HeapInput("SN PI", dci_ti); //not
+      // important in terms of shape (as it will be empty when using)
+      auto fdc =
+          fgb.SenFusedDeviceCompute("FDC", {dci_ti}, {inp_node}, fdc_graph);
+      auto d2h_dci = generate_dci(dev_tensor, host2device);
+      sendnn::NodePtr d2h_dci_node = fgb.SenHostCompute(
+          "Sen2Host-HostPrep", cpu_ti, fdc, "SenDataConvert", d2h_dci);
+
+      fgb.PrimaryOutput("SN PO", d2h_dci_node->OutputPort(0));
+    }
+
+    SEN_THROW_NOK(fgb.Finalize(&exec_graph));
```

Comment:
variable names need a pass in this function

Reply:
@ani300 : we revised the names.

Reply:
there's still a lot of 1 character/2 character/3 character variable names. Please make variable names longer and more descriptive